tests/test_load.py

```diff
@@ -821,12 +821,14 @@ def test_load_to_update(tmpdir, dbs, only_metadata):
 @pytest.mark.parametrize(
     'name, version',
     [
-        (DB_NAME, None),
+        # TODO: find out why this was working before?
+        # (DB_NAME, None),
         (DB_NAME, '1.0.0'),
-        pytest.param(  # database does not exist
-            'does-not-exist', None,
-            marks=pytest.mark.xfail(raises=RuntimeError),
-        ),
+        # TODO: find out why this was working before?
+        # pytest.param(  # database does not exist
+        #     'does-not-exist', None,
```

Comment:
I would replace this with

```suggestion
        #     'does-not-exist', '1.0.0',
```

and then re-enable the test as this should indeed raise an error:

```python
>>> audb.repository('non-existent', '1.0.0')
...
RuntimeError: Cannot find version 1.0.0 for database 'non-existent'.
```

Reply:
Maybe with the new backend implementation we can also improve the error message here to:

```
RuntimeError: Cannot find the database 'non-existent'.
```
?

Reply:
Ok, we now have the following tests two failing tests:

```python
        pytest.param(  # database does not exist
            'does-not-exist',
            '1.0.0',
            marks=pytest.mark.xfail(raises=RuntimeError),
        ),
        pytest.param(  # version does not exist
            DB_NAME,
            '999.9.9',
            marks=pytest.mark.xfail(raises=RuntimeError),
        )
```

> Maybe with the new backend implementation we can also improve the error message here to

So you suggest to raise `RuntimeError: Cannot find the database 'non-existent'` if no version of the database exists, right? And `Cannot find version 999.9.9 for database 'emodb'` if only the requested version is not found, correct?

Reply:
Yes, as this would provide more information to the user. E.g. if you get
```
Cannot find version '1.0.0' for database 'mozillacommmonvoice'
```
You might wonder why version 1.0.0 does not exist instead of realizing that you made a typo in the database name. Hopefully, you find this faster when getting:
```
Cannot find the database 'mozillacommmonvoice'
```

Reply:
Ok, changed it accordingly and updated the test to check for the expected error message.